### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by victor-paumier**
- Improve edit profile on error
- Improve new message mail notification
- fix sort by semesters on matching list
- Allow matching of mentee staying for a year with a mentor only available for a semester
- set default nb elem in lists to 10
- fix deleting thread participant on change user role
- add column visibly button on lists to hide columns dynamically
- popup list buddies on matching summary

### CODE REVIEW

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -90,7 +91,7 @@ public function getListUsers(
     }
 
     /**
-     * @Route("/{id}/edit", name="user_edit")
+     * @Route("/{id}/edit", name="user_edit", options={ "expose" : true })
```
Comment (JeremieSamson) [inline src/Controller/UserController.php]: `, condition="request.isXmlHttpRequest()` ?

File: templates/Buddy/list.html.twig [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<p>
+    {{ 'user.current'|trans ~ ': ' }}
+    <a href="mailto:{{ user.email }}">{{ user.email }}</a>
+</p>
+
+<table class="table table-striped table-hover nowrap custom-datatable" id="list_buddies_user">
+    <thead>
+    <tr class="bg-white text-primary">
```
Comment (JeremieSamson) [inline templates/Buddy/list.html.twig]: indent all tr ?

File: src/Controller/UserController.php (hunk 1 above)
Comment (victor-paumier) [reply src/Controller/UserController.php]: this route must stay accessible directly without ajax too, so no

File: templates/Buddy/list.html.twig (hunk 2 above)
Comment (victor-paumier) [reply templates/Buddy/list.html.twig]: looks coherent with other declaration of datatable, so I would say no need

**Pull request merged**